pkg/cri/resource-manager/metrics/metrics.go

```diff
@@ -193,3 +196,16 @@ func dump(prefix string, f *model.MetricFamily) {
 func metricsError(format string, args ...interface{}) error {
 	return fmt.Errorf("metrics: "+format, args...)
 }
+
+// set the updated policy metrics.
+func (m *Metrics) Set(pm *metrics.PolicyMetrics) {
+	m.Lock()
+	defer m.Unlock()
+	m.policyMetrics = pm
+	data, err := json.Marshal(m.policyMetrics)
+	if err != nil {
+		err = metricsError("failed to marshal policy metrics for monitor: %v", err)
+		return
+	}
+	metrics.Set(string(data))
```

Comment:
nitpick: if you'd store the (locally) opaque policy metrics as `[]byte` instead of a `string` you could avoid the double-conversion to string here and later from string in the collector.